dist/tools/compile_test/compile_like_murdock.py

```diff
@@ -136,37 +136,48 @@ def _end(sec, job):
 
 def _all_apps(cwd):
     cmd = ['make', 'info-applications', '--no-print-directory']
-    out = subprocess.check_output(cmd, cwd=cwd)
+    out = subprocess.check_output(cmd, cwd=cwd, stderr=subprocess.STDOUT)
     return out.decode("utf-8", errors="replace").split()
 
 
 def _supported_boards(boards, env, cwd):
     cmd = ['make', 'info-boards-supported', '--no-print-directory']
-    out = subprocess.check_output(cmd, env=env, cwd=cwd)
+    out = subprocess.check_output(cmd, env=env, cwd=cwd, stderr=subprocess.STDOUT)
     supported_boards = out.decode("utf-8", errors="replace").split()
+    if boards[0] == "all":
```

Comment:
I think this check could be better placed in `main()`, to keep the arg parsing central, and then just pass a flag?
Alternatively, keep this function as the name suggest and always return all supported board, filter the requested boards somewhere else? 
Checking the length of it could also be reasonable, to catch people doing stuff like `-b all,naive,gba`